GVFS/GVFS.Common/GitStatusCacheConfig.cs

```diff
@@ -82,42 +82,8 @@ private static bool TryLoadBackOffTime(GitProcess git, out int backoffTimeSecond
 
         private static bool TryGetFromGitConfig(GitProcess git, string configName, int defaultValue, int minValue, out int value, out string error)
         {
-            value = defaultValue;
-            error = string.Empty;
-
-            GitProcess.Result result = git.GetFromConfig(configName);
-            if (result.HasErrors)
-            {
-                if (result.Errors.Any())
-                {
-                    error = "Error while reading '" + configName + "' from config: " + result.Errors;
-                    return false;
-                }
-
-                // Git returns non-zero for non-existent settings and errors.
-                return true;
-            }
-
-            string valueString = result.Output.TrimEnd('\n');
-            if (string.IsNullOrWhiteSpace(valueString))
-            {
-                // Use default value
-                return true;
-            }
-
-            if (!int.TryParse(valueString, out value))
-            {
-                error = string.Format("Misconfigured config setting {0}, could not parse value {1}", configName, valueString);
-                return false;
-            }
-
-            if (value < minValue)
-            {
-                error = string.Format("Invalid value {0} for setting {1}, value must be greater than or equal to {2}", value, configName, minValue);
-                return false;
-            }
-
-            return true;
+            GitProcess.ConfigResult result = git.GetFromConfig(configName);
+            return result.TryParseAsInt(defaultValue, minValue, out value, out error);

```

Comment:
Previously we'd call `TrimEnd('\n')` on the output here before trying to parse it as an int.  Does `TryParseAsInt` need to do the same?

Reply:
I validated my assumption that int.TryParse() would do that trimming for us. I think it's cleaner to leave that to the framework.